### ISSUE TITLE
List count not correct

### ISSUE THREAD

**Issue by geocachermgo** [cgeo/cgeo#8621]
![Screenshot_2020-07-12-22-49-20-587_cgeo geocaching](https://user-images.githubusercontent.com/43813636/87286381-18313400-c4f9-11ea-9bba-dfe3c15d243c.jpg)
![Screenshot_2020-07-12-22-49-31-679_cgeo geocaching](https://user-images.githubusercontent.com/43813636/87286422-241cf600-c4f9-11ea-82c3-50c7b95543a8.jpg)

**Describe the bug:**
A cache is incorrectly shown a member of an empty list.

I had a cache stored in the list "ReiseLog" and removed it from that list.

Afterwards it was still shown as in this list, despite the list being empty. (Screenshot 1)
After click on the pencil button it showed Screenshot 2, confirming "ReiseLog" having 0 caches.

**Actual behavior/state after performing these steps:**
Describe the actual issue/problem/behavior in detail.

**Version of c:geo used:**

**Is the problem reproducible:**
After opening the empty list and going back to the cache, the problem was not there any more.

**Screenshots:**
You may attach screenshots if applicable and helpful to explain your problem

**System information:**
```
--- System information ---
Device: MI 8 (dipper, Xiaomi)
Android version: 10
Android build: QKQ1.190828.002 test-keys
c:geo version: 2020.07.12-NB-95a79ba
Google Play services: disabled - 20.24.14 (120400-319035315)
Low power mode: active
Compass capabilities: yes
Rotation vector sensor: present
Orientation sensor: present
Magnetometer & Accelerometer sensor: present
Direction sensor used: rotation vector
Hide caches: own/found 
Hide waypoints: -
HW acceleration: enabled (default state)
System language: de_DE
System date format: dd.MM.yy
Debug mode active: no
System internal c:geo dir: /data/user/0/cgeo.geocaching (91,0 GB free) internal
User storage c:geo dir: /storage/emulated/0/cgeo (91,0 GB free) external non-removable
Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (91,0 GB free) external non-removable
Database: /data/user/0/cgeo.geocaching/databases/data (10,0 MB) on system internal storage
Fine location permission: granted
Write external storage permission: granted
Geocaching sites enabled:
   geocaching.com: Logged in (Anmeldung OK) / PREMIUM
Geocaching.com date format: MM/dd/yyyy
Installed c:geo plugins:  none
--- End of system information ---

```

**Additional context:**
e.g. Reference to other issues, projects, sources, etc.

**Comment by Lineflyer** [comment]
You forgot the screenshots.
I assume this could be a refresh problem after modifying the list.

**Comment by geocachermgo** [comment]
I added screenshots yesterday but it seems that it did  not work.

**Comment by Lineflyer** [comment]
Thanks.
Can you provide a step-by-step instruction how to reproduce this (e.g. which function did you use to remove the cache from the list)?

**Comment by geocachermgo** [comment]
1) Create an list with one cache "ReiseLog" and a list with some caches (including) "Germersheim" the one cache
2) Open Cache (Screenshot 1)
3) Click an one cache list "ReiseLog"
4) Menu, Caches verwalten, Alle verschieben, "Germersheim"
5) Report "List is empty, delete" ==> Abort
6) top left corner: back arrow
Then both lists are still mentioned. This is a bug
7) Tap on pencil (Screenshot 2)
Empty list is marked with 0 elements but hook at empty list is set

**Comment by Lineflyer** [comment]
Thanks.
I guess the "trigger point" is, that you accessed the lists from their link inside of a cache, where we miss to update the list information when coming back to cache details page.

**Comment by geocachermgo** [comment]
So is this a known bug?

**Comment by Lineflyer** [comment]
No, just my guess which need to be investigated by someone who is not just guessing. ;)

**Comment by Lineflyer** [comment]
Assigning myself for a retest using the given scenario above.

**Comment by geocachermgo** [comment]
Checked it with 2021.12.21-NB-944fcba. Bug is still present.

**Comment by murggel** [comment]
see PR #16660
